### PR TITLE
[REF]: callback replacement with promises in server.ts

### PR DESCRIPTION
**Pull request by rastajpa**

### CODE REVIEW

File: packages/bitcore-wallet-service/test/integration/server.js [hunk 1]
```diff
@@ -1463,14 +1460,14 @@ describe('Wallet service', function() {
...
-          server.getWallet({}, (err, w) => {
+          server.getWallet({}).then(w => {
 
             var old = server.getWallet;
             server.getWallet = sinon.stub();
 
             // return main address index to 0;
             w.addressManager.receiveAddressIndex = 0;
-            server.getWallet.callsArgWith(1, null, w);
+            server.getWallet.returns(new Promise((resolve) => { resolve(w) }));
```
Comment (micahriggan) [inline packages/bitcore-wallet-service/test/integration/server.js]: you can use yields instead of new promise

**Comment by micahriggan** [comment]
Merge conflicts

File: packages/bitcore-wallet-service/test/integration/server.js (hunk 1 above)
Comment (rastajpa) [reply packages/bitcore-wallet-service/test/integration/server.js]: I think yield is for functions with callbacks... So I replaced the new promise with `resolves`. Let me know if I am wrong

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 2]
```diff
@@ -1464,20 +1473,21 @@ export class WalletService {
...
    * @returns {truthy} The result of the verification.
    */
-  verifyMessageSignature(opts, cb) {
-    if (!checkRequired(opts, ['message', 'signature'], cb)) return;
+  verifyMessageSignature(opts) {
+    return new Promise((resolve) => {
 
-    this.getWallet({}, (err, wallet) => {
-      if (err) return cb(err);
+      if (!checkRequired(opts, ['message', 'signature'])) return;
```
Comment (Gamboster) [inline packages/bitcore-wallet-service/src/lib/server.ts]: Should this `return` be empty?

File: packages/bitcore-wallet-service/test/integration/server.js [hunk 3]
```diff
@@ -697,8 +697,7 @@ describe('Wallet service', function() {
           helpers.getAuthServer(copayerId, function(server) {
             server.getStatus({
               includeExtendedInfo: true
-            }, function(err, status) {
-              should.not.exist(err);
```
Comment (Gamboster) [inline packages/bitcore-wallet-service/test/integration/server.js]: Should be tested a `catch(err => should.not.exist(err))`

**Pull request closed without merging**